src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/settings/ScanTypesPanelTest.kt

```diff
@@ -0,0 +1,147 @@
+package io.snyk.plugin.ui.toolwindow.settings
+
+import UIComponentFinder.getComponentByName
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.util.Disposer
+import com.intellij.openapi.util.registry.Registry
+import com.intellij.testFramework.LightPlatform4TestCase
+import com.intellij.ui.components.JBCheckBox
+import io.mockk.every
+import io.mockk.mockkStatic
+import io.mockk.spyk
+import io.mockk.unmockkAll
+import io.mockk.verify
+import io.snyk.plugin.getKubernetesImageCache
+import io.snyk.plugin.pluginSettings
+import io.snyk.plugin.resetSettings
+import io.snyk.plugin.ui.settings.ScanTypesPanel
+import org.junit.Test
+
+@Suppress("FunctionName")
+class ScanTypesPanelTest : LightPlatform4TestCase() {
+
+    override fun setUp() {
+        super.setUp()
+        unmockkAll()
+        resetSettings(project)
+        disposable = Disposer.newDisposable()
+        // restore modified Registry value
+        isContainerEnabledRegistryValue.setValue(isContainerEnabledDefaultValue)
+    }
+
+    override fun tearDown() {
+        unmockkAll()
+        resetSettings(project)
+        disposable.dispose()
+        // restore modified Registry value
+        isContainerEnabledRegistryValue.setValue(isContainerEnabledDefaultValue)
+        super.tearDown()
+    }
+
+    private lateinit var disposable: Disposable
+
+    private val isContainerEnabledRegistryValue = Registry.get("snyk.preview.container.enabled")
+    private val isContainerEnabledDefaultValue: Boolean by lazy { isContainerEnabledRegistryValue.asBoolean() }
+
+    private fun setUpContainerTest() {
+        isContainerEnabledRegistryValue.setValue(true)
+    }
+
+    @Test
+    fun `container scan enablement get from settings`() {
+        setUpContainerTest()
+        val settings = pluginSettings()
+        settings.containerScanEnabled = true
+
+        val scanTypesPanel = ScanTypesPanel(project, disposable)
+        val containerCheckBox =
+            getComponentByName(scanTypesPanel.panel, JBCheckBox::class, "containerEnablementCheckBox")
+                ?: throw IllegalStateException("containerEnablementCheckBox not found")
+
+        assertTrue("", containerCheckBox.isSelected)
```

Comment:
Should have an error message for the case your checking. Like "Expected container checkbox to be checked from setting, but wasn't"